File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        sort(begin(people), end(people));
        int l = 0, r = size(people) - 1, boats = 0;
        while (l <= r) {
            int weight = people[l] + people[r];
            if (weight <= limit) 
                ++boats, ++l, --r;
            else 
                ++boats, --r;
        }
        return boats;
    }
};

File: 0881-boats-to-save-people/0881-boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        int i = 0, j = people.length - 1, boats = 0;
        Arrays.sort(people);
        while(i <= j) {
            if(people[i] + people[j] <= limit) {
                ++i;
                --j;
            } else --j;
            ++boats;
        }
        return boats;
    }
}